File: vokimi_api/Src/enums/GeneralTestAnswerType.cs
namespace vokimi_api.Src.enums
{
    public enum GeneralTestAnswerType
    {
        TextOnly,
        TextAndImage,
        ImageOnly,
        //color
    }
    public static class GeneralTestAnswerTypeExtensions
    {
        public static bool HasImage(this GeneralTestAnswerType type) => type switch {
            GeneralTestAnswerType.TextOnly => false,
            GeneralTestAnswerType.TextAndImage => true,
            GeneralTestAnswerType.ImageOnly => true,
            _ => throw new NotImplementedException()
        };
        public static string GetId(this GeneralTestAnswerType type) => type switch {
            GeneralTestAnswerType.TextOnly => "text_only",
            GeneralTestAnswerType.TextAndImage => "text_and_image",
            GeneralTestAnswerType.ImageOnly => "image_only",
            _ => throw new NotImplementedException()
        };
        public static GeneralTestAnswerType? FromId(string? id) => id switch {
            "text_only" => GeneralTestAnswerType.TextOnly,
            "text_and_image" => GeneralTestAnswerType.TextAndImage,
            "image_only" => GeneralTestAnswerType.ImageOnly,
            _ => null
        };
    }
}

File: vokimi_frontend/src/ts/enums/GeneralTestAnswerType.ts
export enum GeneralTestAnswerType {
    TextOnly = "text_only",
    TextAndImage = "text_and_image",
    ImageOnly = "image_only"
}

export namespace GeneralTestAnswerTypeUtils {
    export function fromId(id: string): GeneralTestAnswerType {
        for (const key in GeneralTestAnswerType) {
            if (GeneralTestAnswerType[key as keyof typeof GeneralTestAnswerType] === id) {
                return GeneralTestAnswerType[key as keyof typeof GeneralTestAnswerType];
            }
        }
        return GeneralTestAnswerType.TextOnly;
    }

    export function getId(type: GeneralTestAnswerType): string {
        return type;
    }

    export function getFullName(type: GeneralTestAnswerType): string {
        switch (type) {
            case GeneralTestAnswerType.TextOnly:
                return "Text Only";
            case GeneralTestAnswerType.TextAndImage:
                return "Text and Image";
            case GeneralTestAnswerType.ImageOnly:
                return "Image Only";
            default:
                return "Unknown Answer Type";
        }
    }
}
